v0/destinations/freshmarketer/utils.js

```
@@ -0,0 +1,117 @@
+};
+
+/*
+ * This functions is used for getting Account details.
+ * If account is not exists then it will create it otherwise it will update it.
+ * @param {*} payloadBody
+ * @param {*} Config
+ * @returns
+ * ref: https://developers.freshworks.com/crm/api/#upsert_an_account
+ */
+const createUpdateAccount = async (payloadBody, Config) => {
+  const requestOptions = {
+    headers: {
+      Authorization: `Token token=${Config.apiKey}`,
+      "Content-Type": "application/json"
+    }
+  };
+  const endPoint = `https://${Config.domain}${CONFIG_CATEGORIES.GROUP.baseUrl}`;
+  let accountResponse = await httpPOST(endPoint, payloadBody, requestOptions);
+  accountResponse = processAxiosResponse(accountResponse);
+  if (accountResponse.status !== 200) {
+    const errMessage = accountResponse.response.errors?.message || "";
+    const errorStatus = accountResponse.response.errors?.code || "500";
+    throw new CustomError(
+      `failed to create/update group ${errMessage}`,
+      errorStatus
+    );
+  }
+  return accountResponse;
+};
+
+/*
+ * This functions is used for checking Number Data Types of payload.
+ * If the specified key is defined and it is not number then it throws error.
+ * @param {*} payload
+ * @returns
+ */
+const checkNumberDataType = payload => {
+  const numberAttributes = [
+    "territory_id",
+    "lead_source_id",
+    "owner_id",
+    "campaign_id",
+    "contact_status_id",
+    "lifecycle_stage_id",
+    "industry_type_id",
+    "business_type_id",
+    "number_of_employees",
+    "parent_sales_account_id"
+  ];
+  const errorAttributes = [];
+  numberAttributes.forEach(element => {
+    if (payload[element]) {
+      const value = payload[element];
+      if (!isNaN(Number(value))) {
+        payload[element] = Number(value);
+      } else {
+        errorAttributes.push(element);
+      }
+    }
+  });
+  if (errorAttributes.length > 0) {
+    throw new CustomError(`${errorAttributes}: invalid number format`);
```

As none of these fields are required, Is it necessary to throw an error here?

---

If we allow these with wrong data types then Freshmarketer will throw an error. We can ignore these fields if it's wrong, but again it might not be what the customer wants.

---

Adding metadata type : toNumber in configs, it'll ignore the fields if they're `NaN`.

---

Yeah, I discussed this with @ItsSudip, and we found it's better to throw an error, instead of ignoring it.

---

If we silently ignore this field then it will be ignored always. But if we throw an error then there will be a chance from the user's site to rectify the data type.